### ISSUE TITLE
1.16.1: aF - Mvdwplaceholderapi problem with the 8.7.4

### ISSUE THREAD

**Issue by Volgeop** [kangarko/ChatControl-Pro#1817]
Hi, I'm getting this error in console on my 1.16.1 server:
```
[03:00:55] [Server thread/INFO]: ChatControl 8.7.4 encountered an aF! Please check your error.log and report this issue with the information in that file.
[03:00:55] [Server thread/INFO]: org.mineacademy.chatcontrol.final.nul. .aF: Report / Method cannot be null for null
[03:00:55] [Server thread/INFO]: at org.mineacademy.chatcontrol.final.nul. .W.a(SourceFile:56)
[03:00:55] [Server thread/INFO]: at org.mineacademy.chatcontrol.final.nul. .R.a(SourceFile:398)
[03:00:55] [Server thread/INFO]: at org.mineacademy.chatcontrol.lib.fo.model.B.a(SourceFile:1964)
[03:00:55] [Server thread/INFO]: at org.mineacademy.chatcontrol.lib.fo.model.u.b(SourceFile:941)
[03:00:55] [Server thread/INFO]: at org.mineacademy.chatcontrol.lib.fo.model.Y.a(SourceFile:263)
[03:00:55] [Server thread/INFO]: at org.mineacademy.chatcontrol.lib.fo.model.Y.a(SourceFile:227)
[03:00:55] [Server thread/INFO]: at org.mineacademy.chatcontrol.lib.fo.model.Y.a(SourceFile:186)
[03:00:55] [Server thread/INFO]: at org.mineacademy.chatcontrol.lib.fo.model.Y.a(SourceFile:212)
[03:00:55] [Server thread/INFO]: at org.mineacademy.chatcontrol.final.nul. .df.b(SourceFile:377)
[03:00:55] [Server thread/INFO]: at org.mineacademy.chatcontrol.final.nul. .de.a(SourceFile:422)
[03:00:55] [Server thread/INFO]: at org.mineacademy.chatcontrol.final.nul. .dj.a(SourceFile:97)
[03:00:55] [Server thread/INFO]: at org.mineacademy.chatcontrol.final.nul. .di.a(SourceFile:148)
[03:00:55] [Server thread/INFO]: at org.mineacademy.chatcontrol.final.nul. .di.a(SourceFile:136)
[03:00:55] [Server thread/INFO]: at org.mineacademy.chatcontrol.final.nul. .di.a(SourceFile:132)
[03:00:55] [Server thread/INFO]: at org.mineacademy.chatcontrol.final.nul. .da.a(SourceFile:238)
[03:00:55] [Server thread/INFO]: at org.mineacademy.chatcontrol.final.nul. .da.onQuit(SourceFile:288)
[03:00:55] [Server thread/INFO]: at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor715.execute(Unknown Source)
[03:00:55] [Server thread/INFO]: at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69)
[03:00:55] [Server thread/INFO]: at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80)
[03:00:55] [Server thread/INFO]: at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
[03:00:55] [Server thread/INFO]: at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:607)
[03:00:55] [Server thread/INFO]: at org.spigotmc.RestartCommand.shutdownServer(RestartCommand.java:77)
[03:00:55] [Server thread/INFO]: at org.spigotmc.RestartCommand.restart(RestartCommand.java:61)
[03:00:55] [Server thread/INFO]: at org.spigotmc.RestartCommand.restart(RestartCommand.java:40)
[03:00:55] [Server thread/INFO]: at org.spigotmc.RestartCommand$1.run(RestartCommand.java:31)
[03:00:55] [Server thread/INFO]: at java.base/java.lang.Thread.run(Thread.java:832)
```

**Comment by AngellitoX** [comment]
Hello there! Any steps to reproduce the issue?

**Comment by Volgeop** [comment]
Not sure, but I have full console of it. @AngellitoX https://pastebin.com/ydeQPjJU. Happening since the newest .4 version, before it didn't happen.

**Comment by Volgeop** [comment]
Have to downgrade...

**Comment by Volgeop** [comment]
And also server crash on .3 version... nice https://pastebin.com/EvPFPSBq

**Comment by AngellitoX** [comment]
Hello @Volgeop.

There definitely seems to be a problem with your server/plugins because right now I'm running ChatControl 8.7.4 in my 150 players network with absolutely no issues. No crashes not a single similar error log like the one you're showing.

ChatControl shows there because it fails to do its job properly as your server is crashing. Check your timings and look for the source of lag.

Hope this helps!

**Comment by Volgeop** [comment]
But there's mineacademy and null in this server crash. And about version 8.7.4 I already sent you error that also points to something wrong with Chc and the newest ver... on 8.7.3 it doesn't appear.

**Comment by AngellitoX** [comment]
Are you having a overloaded formatting setup? Like too many placeholders, or too large? 

Your timings are definitely crazy, 6 players and crashing. By looking at your timings, InteractionVisualizer and AuctionHouse seems to be consuming too much of your CPU and resources. Try disabling it first and run tests again. 

Also your spawn-limits from bukkit.yml look a bit high for a 1.16 server. Try to reduce that too and keep testing.

Also in the overall consumption, ChatControl looks to be consuming less than 0.15% of usage.
`ChatControl::Combined Totalcount(65896)  total(0.74% 20.997s, 0.57% of tick)`

**Comment by Volgeop** [comment]
I already talked with these two devs and ChatControl seems to be more heavy (red numbers) than these two. They don't think it's too critical.

```
spawn-limits:
  water-ambient: 20
  monsters: 30
  animals: 8
  water-animals: 3
  ambient: 1
```

This is high for you? What do you suggest me?


Btw my Chc format file.
https://pastebin.com/n1iYxnac

**Comment by AngellitoX** [comment]
Wow, you are using almost 20 placeholders in your format. That is definitely your issue.

Think about it this way: everytime a user chats, each of the 20 placerholder needs to ask and collect the data to provide the accurate information in the final output. Some of those placeholders could also be hardcoded which is also making your server lag because it takes a lot of time to retrieve the information.

As stated on the Formatting sections from the wiki, the more placeholders you use, the more performance impact you will get.

Also, the performance impact will also depend on your host specs. The better they are, the more placeholders you will be able to support.

**Comment by Volgeop** [comment]
And do you think this is my issue? How many placeholders do you suggest me? Which one do you suggests remove?

But about error from 8.7.4, it's probably problem in this version, when it doesn't happen on previous 8.7.3, no?

**Comment by AngellitoX** [comment]
I would recommend to have maximum half of the placeholders that you have right now. I can't say which one is better to remove because it depends on your server and what's most important to show for you.

First of all, try to reduce the huge amount of placeholders and check if that solves your current problem.

As for 8.7.4 I can't really say bro. As I told you, I am using it for 12hrs now with 150 players without any major problem.

**Comment by Volgeop** [comment]
BUt I sent you error, did it seen Kangarko? He may know how to solve this error.

**Comment by AngellitoX** [comment]
The error can be also caused by the same overload of placeholders. Once again, reduce them first. If your issue persists, then we could check more. But I can confirm that 8.7.4 is stable enough and no errors are appearing to me.

**Comment by Volgeop** [comment]
But btw... I'm using also featherboard and there are placeholders updating also very often?

**Comment by AngellitoX** [comment]
FeatherBoard in my own personal opinion has a reputation of consuming to much CPU in the background. I personally use TAB Premium.

And yes, I dont think you need to show your users the country they belong. Ping is fine afaik.

Try best to disable them all and start enabling tjem one by one.

**Comment by Volgeop** [comment]
And how can I check if the perfomance is better? COuld I debug it somehow?

**Comment by AngellitoX** [comment]
Of course. Timings will always give you the hint. I am pretty sure that by removing the placeholders you will be fine.

**Comment by Volgeop** [comment]
https://hastebin.com/ekazayohoh.php Is this better? Would you suggest remove something more?

**Comment by Volgeop** [comment]
@AngellitoX the 8.7.4 problem are mvdwplaceholders from mvdwplaceholderapi.... But I need this plugin for featherboard. In Chatcontrol I'm using PAPI placeholders, so wtf? Look again at messages in error of 8..7.4, it says it.

**Comment by hwrichardson999** [comment]
Hi there. I have the identical issue
`09.08 14:17:29 [Server] INFO ChatControl 8.7.4 encountered an aF! Please check your error.log and report this issue with the information in that file.
09.08 14:17:29 [Server] INFO org.mineacademy.chatcontrol.final.nul. .aF: Report / Method cannot be null for null
09.08 14:17:29 [Server] INFO at org.mineacademy.chatcontrol.final.nul. .W.a(SourceFile:56)
09.08 14:17:29 [Server] INFO at org.mineacademy.chatcontrol.final.nul. .R.a(SourceFile:398)
09.08 14:17:29 [Server] INFO at org.mineacademy.chatcontrol.lib.fo.model.B.a(SourceFile:1964)
09.08 14:17:29 [Server] INFO at org.mineacademy.chatcontrol.lib.fo.model.u.b(SourceFile:941)
09.08 14:17:29 [Server] INFO at org.mineacademy.chatcontrol.lib.fo.model.Y.a(SourceFile:263)
09.08 14:17:29 [Server] INFO at org.mineacademy.chatcontrol.lib.fo.model.Y.a(SourceFile:227)
09.08 14:17:29 [Server] INFO at org.mineacademy.chatcontrol.lib.fo.model.Y.a(SourceFile:186)
09.08 14:17:29 [Server] INFO at org.mineacademy.chatcontrol.lib.fo.model.Y.a(SourceFile:212)
09.08 14:17:29 [Server] INFO at org.mineacademy.chatcontrol.final.nul. .dL.a(SourceFile:389)
09.08 14:17:29 [Server] INFO at org.mineacademy.chatcontrol.final.nul. .da.b(SourceFile:184)
09.08 14:17:29 [Server] INFO at java.lang.Thread.run(Thread.java:748)
09.08 14:17:29 [Server] INFO !-----------------------------------------------------!
09.08 14:17:29 [Server] INFO MvdWPlaceholders placeholders failed!
09.08 14:17:29 [Server] INFO Player: Alfric_
09.08 14:17:29 [Server] INFO Message: '-----------------------------------------------------
09.08 14:17:29 [Server] INFO Welcome to Factions, {player}!
09.08 14:17:29 [Server] INFO This server is still in development and still
09.08 14:17:29 [Server] INFO has a lot of features to come.
09.08 14:17:29 [Server] INFO -----------------------------------------------------'
09.08 14:17:29 [Server] INFO Consider writing to developer of that library
09.08 14:17:29 [Server] INFO first as this may be a bug we cannot handle!
09.08 14:17:29 [Server] INFO Your chat message will appear without replacements.
09.08 14:17:29 [Server] INFO !-----------------------------------------------------!`

**Comment by Faicu** [comment]
confirm this issue, having same problem, downgraded to .3

**Comment by AngellitoX** [comment]
It looks like Mvdw may be using an event which is not compatible with PAPI installed. Might want to open a ticket to them? I cannot seem to replicate the issue.

**Comment by Volgeop** [comment]
But when it's happening just with 8.7.4 version? On 8.7.3 everything works... I really don't think it's mvdw problem, even here is chatcontrol error. Could kangarko look on it?

**Comment by AngellitoX** [comment]
On 8.7.4 Matej updated the way placeholders work with the new API, so probably something changed. And yes, he can take a look this week.

**Comment by TNTUP** [comment]
I can confirm this issue. Ended up removing the MVDWPlaceholderAPI from my servers, removed AnimatedNames aswell

**Comment by AngellitoX** [comment]
Hello, I was testing last night and I was also able to confirm this too. Looks like Mvdw is using an outdated method. Hope @kangarko can take a look to this soon and see if there's any workaround!

**Comment by iProMC** [comment]
yeah, I have the same issue too. IDK why

**Comment by broken1arrow** [comment]
I get it every time some one login. 

https://pastebin.com/ks0cmz7V

No errors if I reload the plugin, only players join (but it seams it have no issue in game).

11.08 16:02:19 [Server] INFO Message: '⚜~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~⚜
11.08 16:02:19 [Server] INFO Hej {player}, Välkommen tillbaka
11.08 16:02:19 [Server] INFO Uppdateringar på Midgardsaga
11.08 16:02:19 [Server] INFO 2020-05-31
11.08 16:02:19 [Server] INFO ⇝ Demoner
11.08 16:02:19 [Server] INFO ⇝ Ny gui för Miniblock/huvuden
11.08 16:02:19 [Server] INFO Hjälp - /hjälp eller kontakta en Admin/Mod
11.08 16:02:19 [Server] INFO Rösta gärna: xxxxxxxxxxxxxxxxxxx
11.08 16:02:19 [Server] INFO Discord: xxxxxxxxxxxxxxxxxxxxxxxx
11.08 16:02:19 [Server] INFO Karta över världen: xxxxxxxxxxxxxxxxxxxxxx
11.08 16:02:19 [Server] Startup Donera till servern: xxxxxxxxxxxxxxxxxxxxxxxxxxx
11.08 16:02:19 [Server] INFO ⚜~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~⚜'
11.08 16:02:19 [Server] INFO Consider writing to developer of that library
11.08 16:02:19 [Server] INFO first as this may be a bug we cannot handle!
11.08 16:02:19 [Server] INFO Your chat message will appear without replacements.
11.08 16:02:19 [Server] INFO !-----------------------------------------------------!
11.08 16:02:19 [Server] INFO ChatControl 8.7.5 encountered an aF! Please check your error.log and report this issue with the information in that file.
11.08 16:02:19 [Server] INFO org.mineacademy.chatcontrol.final.nul. .aF: Report / Method cannot be null for null
11.08 16:02:19 [Server] INFO at org.mineacademy.chatcontrol.final.nul. .W.a(SourceFile:56)
11.08 16:02:19 [Server] INFO at org.mineacademy.chatcontrol.final.nul. .R.a(SourceFile:398)
11.08 16:02:19 [Server] INFO at org.mineacademy.chatcontrol.lib.fo.model.B.a(SourceFile:1967)
11.08 16:02:19 [Server] INFO at org.mineacademy.chatcontrol.lib.fo.model.u.b(SourceFile:941)
11.08 16:02:19 [Server] INFO at org.mineacademy.chatcontrol.lib.fo.model.Y.a(SourceFile:263)
11.08 16:02:19 [Server] INFO at org.mineacademy.chatcontrol.lib.fo.model.Y.a(SourceFile:227)
11.08 16:02:19 [Server] INFO at org.mineacademy.chatcontrol.lib.fo.model.Y.a(SourceFile:186)
11.08 16:02:19 [Server] INFO at org.mineacademy.chatcontrol.lib.fo.model.Y.a(SourceFile:212)
11.08 16:02:19 [Server] INFO at org.mineacademy.chatcontrol.final.nul. .da.a(SourceFile:386)
11.08 16:02:19 [Server] INFO at org.mineacademy.chatcontrol.final.nul. .da.a(SourceFile:236)
11.08 16:02:19 [Server] INFO at org.mineacademy.chatcontrol.final.nul. .da.a(SourceFile:212)
11.08 16:02:19 [Server] INFO at java.base/java.lang.Thread.run(Thread.java:834)
11.08 16:02:19 [Server] INFO !-----------------------------------------------------!
11.08 16:02:19 [Server] INFO MvdWPlaceholders placeholders failed!

**Comment by AngellitoX** [comment]
Hello everyone.

This issue has been addressed and patched in 8.7.6. Version is already on SpigotMC. Could anybody please test it out and let us know if it was correctly fixed?

If so, I will close the ticket.

Thank you everyone who were patiently waiting for this!

**Comment by Volgeop** [comment]
Yeah, works.

**Comment by broken1arrow** [comment]
> 
> 
> Hello everyone.
> 
> This issue has been addressed and patched in 8.7.6. Version is already on SpigotMC. Could anybody please test it out and let us know if it was correctly fixed?
> 
> If so, I will close the ticket.
> 
> Thank you everyone who were patiently waiting for this!

Yes, it works. See it it show up again.
